### PR TITLE
elasticsearch: Upgrade to Elasticsearch 7.0.1 with minor config mods

### PR DESCRIPTION
**Pull request by essandess**
elasticsearch: Upgrade to Elasticsearch 7.0.1 with minor config mods

* Upgrade to Elasticsearch 7.0.1
* Add notes
* Put ES data in ${prefix}/var/${name}
* Configure for a reliable, small single node cluster

cc: @GerardSoleCa

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14
Xcode 8.latest

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: databases/elasticsearch/Portfile [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
-PortSystem          1.0
+PortSystem                                      1.0
```
Comment (cjones051073) [inline databases/elasticsearch/Portfile]: unnecessary whitespace addition ?

File: databases/elasticsearch/Portfile [hunk 2]
```diff
@@ -15,31 +16,35 @@ long_description    Elasticsearch is a highly scalable open-source full-text \
...
+
+checksums           rmd160  c50a7a2e66e53b5df70afde3da2f971d1ac73886 \
+                    sha256  283859570434cd64146c61e5263c2cb3ce537e0dec567c39e0bc41e507eefed1 \
+                    size    338459414
 
-use_zip             yes
 use_configure       no
 build {}
 
+depends_run         port:openjdk12
```
Comment (cjones051073) [inline databases/elasticsearch/Portfile]: please use the java PortGroup, rather than hardcode this dependency. See for instance

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/java-1.0.tcl

In this case

```
PortGroup        java         1.0
# Required java version.
java.version          12
# JDK port to install if required java not found
java.fallback         openjdk12
```

is probably all you need.

File: databases/elasticsearch/Portfile (hunk 1 above)
Comment (essandess) [reply databases/elasticsearch/Portfile]: Relic of attempt to use github portgroup. Those releases require a full compile, and it’s easier just to download the macOS release directly from elastic.

File: databases/elasticsearch/Portfile (hunk 2 above)
Comment (essandess) [reply databases/elasticsearch/Portfile]: What exactly does this do? After the licensing and compatibility issues with Oracle’s JDK v12, I’ve personally done a complete purge and moved everything entirely to openjdk12. That’s apparently what Elastic does because they bundle an openjdk12 within this release and their script adds it as `JAVA_HOME` as the default.

Because MP already has an `openjdk12`, I included this port as a dependency and omitted the copy that comes bundled within Elastic’s release.

Please advise.

Comment (cjones051073) [reply databases/elasticsearch/Portfile]: It allows for the possibility that the user may have installed it themselves, outside macports, and in that case will use it. Macports port will then only be installed if it is not already there.

Comment (essandess) [reply databases/elasticsearch/Portfile]: Thank you. Elastic’s support matrix says that they support both equally, so your mod is perfect. I will push it.

**Comment by cjones051073** [comment]
Thanks for the update. @GerardSoleCa are you happy for the changes here to be merged ?

**Comment by cjones051073** [comment]
ping @GerardSoleCa  ? Lets give it a few more days then merge under openmaintainer.

**Comment by GerardSoleCa** [comment]
@cjones051073 Go ahead, I'm happy with them. If you want, you can also take over the maintenance of the port, we no longer use it in the company, so it is hard for me to keep it updated.

Thanks!

**Comment by cjones051073** [comment]
@GerardSoleCa If you wish to give up maintainership, please just submit a PR setting it to `nomaintainer`. If @essandess then wishes to pick it up and maintain it,that is up to him.

**Comment by mf2k** [comment]
@GerardSoleCa: If you do/can not be maintainer for this port, just let me know by replying and I will remove you as maintainer.
